samples/snippets/csharp/VS_Snippets_CLR/conceptual.disposable/cs/derived1.cs

```diff
@@ -7,144 +7,162 @@
-   protected const uint OPEN_ALWAYS = 4;
-
-   // Define additional APIs.
-   [DllImport("kernel32.dll")]
-   protected static extern bool WriteFile(
-                                SafeFileHandle safeHandle, string lpBuffer,
-                                int nNumberOfBytesToWrite, out int lpNumberOfBytesWritten,
-                                IntPtr lpOverlapped);
-
-   // Define locals.
-   private bool disposed = false;
-   private string filename;
-   private bool created = false;
-   private SafeFileHandle safeHandle;
-
-   public DisposableStreamResource2(string filename) : base(filename)
-   {
-      this.filename = filename;
-   }
-
-   public void WriteFileInfo()
-   {
-      if (! created) {
-         IntPtr hFile = CreateFile(@".\FileInfo.txt", GENERIC_WRITE, 0,
-                                   IntPtr.Zero, OPEN_ALWAYS,
-                                   FILE_ATTRIBUTE_NORMAL, IntPtr.Zero);
-         if (hFile != INVALID_HANDLE_VALUE)
-            safeHandle = new SafeFileHandle(hFile, true);
-         else
-            throw new IOException("Unable to create output file.");
-
-         created = true;
-      }
-
-      string output = String.Format("{0}: {1:N0} bytes\n", filename, Size);
-      int bytesWritten;
-      bool result = WriteFile(safeHandle, output, output.Length, out bytesWritten, IntPtr.Zero);
-   }
-
-   protected override void Dispose(bool disposing)
-   {
-      if (disposed) return;
-
-      // Release any managed resources here.
-      if (disposing)
-         safeHandle.Dispose();
-
-      disposed = true;
-
-      // Release any unmanaged resources not wrapped by safe handles here.
-
-      // Call the base class implementation.
-      base.Dispose(disposing);
-   }
+    // Define additional constants.
+    protected const uint GENERIC_WRITE = 0x40000000;
+    protected const uint OPEN_ALWAYS = 4;
+
+    // Define additional APIs.
+    [DllImport("kernel32.dll")]
+    protected static extern bool WriteFile(
+        SafeFileHandle safeHandle, string lpBuffer,
+        int nNumberOfBytesToWrite, out int lpNumberOfBytesWritten,
+        IntPtr lpOverlapped);
+
+    bool _disposed = false; // To detect redundant calls
+    bool _created = false;
+    SafeFileHandle _safeHandle;
+    readonly string _fileName;
+
+    public DisposableStreamResource2(string fileName) : base(fileName) => _fileName = fileName;
+
+    public void WriteFileInfo()
+    {
+        if (!_created)
+        {
+            IntPtr hFile = CreateFile(
+                @".\FileInfo.txt", GENERIC_WRITE, 0, IntPtr.Zero,
+                OPEN_ALWAYS, FILE_ATTRIBUTE_NORMAL, IntPtr.Zero);
+
+            if (hFile == INVALID_HANDLE_VALUE)
+            {
+                throw new IOException("Unable to create output file.");
+            }
+
+            _safeHandle = new SafeFileHandle(hFile, true);
+            _created = true;
+        }
+
+        var output = $"{_fileName}: {Size:N0} bytes\n";
```

Comment:
If the samples are meant to follow the BCL style, then `var` shouldn't be used unless the type name is already on the line from a `new`, `as ...` or hard cast (`(...)`).  (`var buf = new byte[50];` is OK, but it's `string txt = string.Empty;` and `string contents = await reader.ReadToEnd();`... here it should use the type name explicitly)